api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -304,6 +306,43 @@
    * @since 0.1.0
    */
   interface Builder {
+    /**
+     * Sets the parent {@code Span} to use. If not set, {@code Tracer.getCurrentSpan()} will be used
+     * as parent.
+     *
+     * <p>This <b>must</b> be used to create a {@code Span} when manual Context propagation is used
+     * OR when creating a root {@code Span} with a {@code null} parent or a parent with an invalid
+     * {@link SpanContext}.
+     *
+     * <p>Observe this is the preferred method when the parent is a {@code Span} created within the
+     * process. Using its {@code SpanContext} as parent remains as a valid, albeit inefficient,
+     * operation.
+     *
+     * <p>If called multiple times, only the last specified value will be used.
+     *
+     * @param parent the {@code Span} used as parent.
+     * @return this.
+     * @since 0.1.0
+     */
+    Builder setParent(@Nullable Span parent);
+
+    /**
+     * Sets the parent {@link SpanContext} to use. If not set, {@code Tracer.getCurrentSpan()} will
+     * be used as parent.
+     *
+     * <p>This <b>must</b> be used to create a {@code Span} when the parent is in a different
+     * process. This is only intended for use by RPC systems or similar.
+     *
+     * <p>If no {@link SpanContext} is available, users must call this method with a {@code null}
```

Comment:
Can you add a test for setParent(null). I think that will not compile.

Reply:
I don't follow this - why MUST users call this with null?  If the intention is to force-start a new trace rather than use currentSpan, then I would prefer some other approach than setParent(null), maybe a default SpanContext constant.

Also, how is `setParent(null)` even going to compile given that the method is overloaded? You'd have to do something like `setParent((Span) null)`

Reply:
That's correct, that won't compile - See the OT shim part, as I had to do `setParent((Span)null))`. This makes me thing we should have totally have them with different names, i.e. `setRemoteParent(SpanContext)`

> I don't follow this - why MUST users call this with null?

Actually that's something that nice to have, similar to the `ignoreActiveSpan()` we had in OT. We can call this `ignoreCurrentSpan()`?

Reply:
I like more `.newTrace()` than `ignoreCurrentSpan()`. It seems more semantically correct if we want to provide an API to start a new trace.

Reply:
So personally I'd like to see `setRootSpan()` or `setAsRootSpan()` to keep the style. Anyway, lets comment about that in the comments section ;)

Reply:
I like `newTrace()` too

Reply:
Thanks for the feedback. So, you might not like it, but trying to keep the current convention, we ought to use `setNewTrace()`. Works for you?

Reply:
setNoParent()? This will also make sense for Tags/Baggages.